docs/api/manual/UseCases.md

```diff
@@ -308,97 +290,97 @@ After the receiver is done with the message, it must free resources backing the
+Likewise, using the process stack for thread stacks is also not feasible. This is because multiple threads cannot share the same stack if they are not guaranteed to stack and unstack in a consistent order (first in, first out). Consider the situation where a thread performs a secure gateway to another process, and then yields and gets swapped out by the RTOS, saving its registers on the process stack. Next, a different thread performs a secure gateway to this same process, and it also yields and gets swapped out. Now, the RTOS wants to swap back in the original thread. The RTOS uses the stack pointer for that thread, and the registers are restored without any issue. The original thread then does a stack allocation that blaps over the saved state of the second thread, and the original thread then yields and is swapped out. Finally, the RTOS wants to swap the second thread back in. It uses the stack pointer for the second thread, but this pointer is now pointed to some "garbage" data leftover from the first thread's stack allocation.
 
-By naively reimplementing the traditional sort of secure gateway in terms of uVisor IPC, we would lose out on some of the static verifiability promises of the original uVisor.
+By naively reimplementing the traditional sort of secure gateway in terms of uVisor IPC, you lose some of the static verifiability promises of the original uVisor.
 
 Previously, a secure gateway stored the following information in flash.
-  - uVisor Secure Gateway Magic (to identify a potential secure gateway)
-  - Target Box (Config Pointer)
-  - Target Function
+  - uVisor Secure Gateway Magic (to identify a potential secure gateway).
+  - Target Box (Config Pointer).
+  - Target Function.
 
 To ensure that such static verifiabiillty promises are maintained, the secure gateway is enhanced to include the following information in flash.
-  - IPC uVisor Secure Gateway Magic (to identify a potential IPC secure gateway)
-  - Source Box (Config Pointer)
-  - Target Box Queue Name
-  - Target Box Namespace (optional)
+  - IPC uVisor Secure Gateway Magic (to identify a potential IPC secure gateway).
+  - Source Box (Config Pointer).
+  - Target Box Queue Name.
+  - Target Box Namespace (optional).
 
-Secure gateway calls are synchronous, to avoid changing semantics. When the target process completes the function call on behalf of the caller, the source process receives the return code for the function call. Complex types (such as a struct) cannot be returned via a secure gateway call.
+Secure gateway calls are synchronous, to avoid changing semantics. When the target process completes the function call on behalf of the caller, the source process receives the return code for the function call. Secure gateway calls cannot return complex types, such as structs.
 
 An asynchronous secure gateway is also introduced, to make asynchronous RPC easier than setting up a queue and using `uvisor_ipc_send`.
 -->
 
```

Comment:
Is this whole section supposed to be hidden? If so, should we find a way to hide it properly, so users can't see it?